arch/risc-v/src/common/riscv_exception_common.S

```diff
@@ -95,7 +101,11 @@ exception_common:
   /* Offset to hartid */
 
   mv         s0, a0               /* save cause */
+#ifdef CONFIG_SMP
```

Comment:
This will change entirely in #5985 so no worries.

Reply:
I've just taken a look.  Thanks for doing that!  It's in line with other archs now.

Do you want me to drop this commit and wait for your PR to be merged?  My rebasing will be without conflict.

Reply:
No no please merge your stuff in first if you get approval, I'll do the fixing on monday so no need to wait for me